src/PanelsView.tsx

```diff
@@ -0,0 +1,129 @@
+import { EditorPanel } from './EditorPanel';
+import { EventsPanel } from './EventsPanel';
+import { Login } from './Login';
+import { ResizableBox } from './ResizableBox';
+import { SettingsPanel } from './SettingsPanel';
+import { useSimulation } from './SimulationContext';
+import { useSourceActor } from './sourceMachine';
+import { SpinnerWithText } from './SpinnerWithText';
+import { StatePanel } from './StatePanel';
+
+export const PanelsView = () => {
+  const simService = useSimulation();
+  const authService = useAuth();
+  const [sourceState, sendToSourceService] = useSourceActor(authService);
+
+  return (
+    <ResizableBox gridArea="panels" minHeight={0}>
+      <Tabs
+        bg="gray.800"
+        display="grid"
+        gridTemplateRows="3rem 1fr auto"
+        height="100%"
+      >
+        <TabList>
+          <Tab>Code</Tab>
+          <Tab>State</Tab>
+          <Tab>Events</Tab>
+          <Tab>Actors</Tab>
+          <Tab marginLeft="auto" marginRight="2">
+            <SettingsIcon />
+          </Tab>
+          <Login />
+        </TabList>
+
+        <TabPanels minHeight={0}>
+          <TabPanel height="100%" padding={0}>
+            {sourceState.matches({
+              with_source: 'loading_content',
+            }) && (
+              <SpinnerWithText
+                text={`Loading source from ${sourceState.context.sourceProvider}`}
+              />
+            )}
+            {!sourceState.matches({
+              with_source: 'loading_content',
+            }) && (
+              <EditorPanel
+                onChangedCodeValue={(code) => {
+                  sendToSourceService({
+                    type: 'CODE_UPDATED',
+                    code,
+                    sourceID: sourceState.context.sourceID,
+                  });
+                }}
+                onCreateNew={() =>
+                  sendToSourceService({
+                    type: 'CREATE_NEW',
+                  })
+                }
+                onSave={() => {
+                  sendToSourceService({
+                    type: 'SAVE',
+                  });
+                }}
+                onChange={(machines) => {
+                  simService.send({
+                    type: 'MACHINES.REGISTER',
+                    machines,
+                  });
+                }}
+                onFork={() => {
+                  sendToSourceService({
+                    type: 'FORK',
+                  });
+                }}
+              />
+            )}
+          </TabPanel>
+          <TabPanel height="100%" overflowY="auto">
+            <StatePanel />
+          </TabPanel>
+          <TabPanel height="100%" overflow="hidden">
+            <EventsPanel />
+          </TabPanel>
+          <TabPanel height="100%" overflowY="auto">
+            <ActorsPanel />
+          </TabPanel>
+          <TabPanel height="100%" overflowY="auto">
+            <SettingsPanel />
+          </TabPanel>
+        </TabPanels>
+        <Box
```

Comment:
I like the current version better but I still feel like this looks a little bit off:
<img width="573" alt="Screenshot 2021-08-19 at 13 39 57" src="https://user-images.githubusercontent.com/9800850/130062553-188ac2bb-eba0-4837-a543-3d3f70a9d6c4.png">

Maybe it's just me though. It's just that when main buttons when aligned to the bottom of the screen it looked visually better for me.